packages/trader/src/AppV2/Components/TradeParameters/Barrier/barrier-input.tsx

```diff
@@ -39,8 +46,29 @@ const BarrierInput = observer(
             } else {
                 setOption(2);
             }
+            onChange({ target: { name: 'barrier_1', value: barrier_1 } });
         }, []);
 
+        React.useEffect(() => {
+            const barrier_element = barrier_ref.current;
+            const checkFocus = () => {
+                setIsFocused(!!(barrier_element && barrier_element.contains(document.activeElement)));
+            };
+            document.addEventListener('focusin', checkFocus);
+            document.addEventListener('focusout', checkFocus);
+
+            return () => {
+                document.removeEventListener('focusin', checkFocus);
+                document.removeEventListener('focusout', checkFocus);
+            };
+        });
```

Comment:
If you do not pass a dependency array to the useEffect hook, it will execute the effect after every render of the component. Is it desired behaviour? Or you wanted to pass empty array?

Reply:
Yeah, this is the intended behavior. It doesn't work correctly if I pass empty array, I need to listen for every component updates